### PR TITLE
feat: cheat window

### PR DESCRIPTION
**Pull request by LPLafontaineB**
### Description (*)

This PR replaces the keyboard hotkeys with a popup window for debug cheats. The window can be opened or closed by the host or a client, using the `/` key, or with 4 fingers on a mobile device. The three cheats that we currently have can now be accessed via buttons.
The window and its cheats are only accessible in-editor and in development builds.

DebugCheatsPopupWindowV2

### Issue Number(s) (*)

Fixes issue(s): [MTT-1994](https://jira.unity3d.com/browse/MTT-1994)
### Manual testing scenarios

1. Start a game
2. Select a character and go to BossRoom scene
3. On host or client, press `/` key to open the debug cheats window
4. Press any button to test the corresponding cheat
### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: Assets/BossRoom/Prefabs/UI/BossRoomHudCanvas.prefab
Comment (fernando-cortez) [inline Assets/BossRoom/Prefabs/UI/BossRoomHudCanvas.prefab:1286]: Just a general nitpicky cleanup comment: for UI elements, keep "Raycast Target" enabled only for the necessary components. Some text items, and nested images can have that field disabled. TextMeshPro components have these hidden under "Extra Settings" btw.

File: Assets/BossRoom/Scripts/Client/UI/UIDebugCheats.cs
Comment (LPLafontaineB) [reply Assets/BossRoom/Scripts/Client/UI/UIDebugCheats.cs]: Great idea! It is much cleaner now, and I could remove the DebugCheatsMediator altogether!

File: Assets/BossRoom/Scripts/Client/UI/UIDebugCheats.cs
Comment (fernando-cortez) [inline Assets/BossRoom/Scripts/Client/UI/UIDebugCheats.cs:2]: Can remove now 👍.

**Review by fernando-cortez — COMMENTED** [review]
Just needs an unused namespace removed, but LGTM!

**Review by fernando-cortez — DISMISSED** [review]

File: Assets/BossRoom/Scripts/Server/DebugCheatsManager.cs
Comment (SamuelBellomo) [inline Assets/BossRoom/Scripts/Server/DebugCheatsManager.cs:35]: do you need those wrappers? Couldn't you call the RPCs themselves?

File: Assets/BossRoom/Scripts/Server/DebugCheatsManager.cs
Comment (SamuelBellomo) [inline Assets/BossRoom/Scripts/Server/DebugCheatsManager.cs:38]: Since your call sites for these RPCs don't use serverRpcParams, I believe you can just remove "default"

File: Assets/BossRoom/Scripts/Server/DebugCheatsManager.cs
Comment (SamuelBellomo) [inline Assets/BossRoom/Scripts/Server/DebugCheatsManager.cs:61]: (not a review comment) lol the snitching method :P

File: Assets/BossRoom/Scripts/Client/UI/UIDebugCheats.cs
Comment (SamuelBellomo) [inline Assets/BossRoom/Scripts/Client/UI/UIDebugCheats.cs:14]: My understanding is you're putting all those events in between because of the client/server separation correct?
What do you think of putting everything in Shared and hooking up the buttons directly to the RPCs? It'd make things way simpler and we could put this cheat code in a self contained "cheat assembly", something like that

File: Assets/BossRoom/Scripts/Server/DebugCheatsManager.cs
Comment (LPLafontaineB) [reply Assets/BossRoom/Scripts/Server/DebugCheatsManager.cs:35]: The `UnityEventGameEventListener`s uses parameterless `UnityEvent`s, so I could not use the RPC calls directly

File: Assets/BossRoom/Scripts/Server/DebugCheatsManager.cs
Comment (LPLafontaineB) [reply Assets/BossRoom/Scripts/Server/DebugCheatsManager.cs:38]: They do use it, for the SenderClientId. This is used for the "snitch log", but it is required anyways for the god mode cheat, if we want to apply it to only a single player.

File: Assets/BossRoom/Scripts/Client/UI/UIDebugCheats.cs
Comment (LPLafontaineB) [reply Assets/BossRoom/Scripts/Client/UI/UIDebugCheats.cs:14]: > My understanding is you're putting all those events in between because of the client/server separation correct?
Yes, that and to keep the UI and logic separate.

> What do you think of putting everything in Shared and hooking up the buttons directly to the RPCs?
It would work for the cheats that are in this PR, but we would lose access to everything in the Server assembly. This could be a problem for some new cheats as well. For example, my current implementation of God Mode uses PlayerServerCharacter to find the character on which to apply the toggle. I could move it to a new assembly with a dependency on Server and Shared, though. It would remove the need for events

File: Assets/BossRoom/Scripts/Server/DebugCheatsManager.cs
Comment (SamuelBellomo) [reply Assets/BossRoom/Scripts/Server/DebugCheatsManager.cs:35]: I see, that's how C# method signature works :( 
nvm

File: Assets/BossRoom/Scripts/Server/DebugCheatsManager.cs
Comment (SamuelBellomo) [reply Assets/BossRoom/Scripts/Server/DebugCheatsManager.cs:38]: aaand that's what happens when you read too fast.
nvm

**Review by SamuelBellomo — DISMISSED** [review]

**Comment by LPLafontaineB** [comment]
Updated PR description with screenshot of new window with instructions, which is now active at the start of the game.

**Review by SamuelBellomo — APPROVED** [review]

**Review by fernando-cortez — COMMENTED** [review]
Just needs StreamingAssets meta file removed, but gtg.

**Review by fernando-cortez — APPROVED** [review]

**Review by SamuelBellomo — APPROVED** [review]

**Pull request merged**
